Modules/System/Email/src/Email/Email.Codeunit.al

```diff
@@ -239,6 +239,17 @@ codeunit 8901 "Email"
         exit(EmailImpl.GetSentEmailsForRecord(TableId, SystemId));
     end;
 
+    ///<summary>
+    /// Gets the outbox emails related to a record.
+    ///</summary>
+    ///<param name="TableId">The table ID of the record.</param>
+    ///<param name="SystemId">The system ID of the record.</param>
+    ///<returns>The outbox emails related to a record.</returns>
+    procedure GetEmailOutboxForRecord(TableId: Integer; SystemId: Guid) ResultEmailOutbox: Record "Email Outbox" temporary;
```

Comment:
This results in copying the result record to a new instance and unfortunately this only copies the fields and filters not the recordset. There is a nice [blogpost ](https://www.waldo.be/2021/07/06/the-complexity-of-complex-return-types/) that describes the issue in depth

Reply:
initially, I have created a copy of **GetSentEmailsForRecord** function, with only difference in the resulting record.
I totally agree with your idea that it would be better to return results by reference, so I have created additional overload for **GetEmailOutboxForRecord** and **GetSentEmailsForRecord** procedures in Email Codeunit to keep the same logic for both procedures

Reply:
I have removed 

` GetEmailOutboxForRecord(TableId: Integer; SystemId: Guid) ResultEmailOutbox: Record "Email Outbox" temporary;`

and changed procedure GetEmailOutboxForRecord to use Variant instead of RecordRef: 
`GetEmailOutboxForRecord(RecordVariant: Variant; var ResultEmailOutbox: Record "Email Outbox" temporary);`